kernel/src/transaction/mod.rs

```diff
@@ -240,6 +319,99 @@ impl Transaction {
     pub fn add_files(&mut self, add_metadata: Box<dyn EngineData>) {
         self.add_files_metadata.push(add_metadata);
     }
+
+    /// Provide a list of files to remove from the table in this transaction. This API together
+    /// with add_files allows engines to delete and update rows at the table at a file level
+    /// granularity.
+    ///
+    /// The metadata passed in here is expected to be taken from the output of a scan of the
+    /// table.
+    /// The expected schema for `remove_metadata` is given by [`crate::scan::scan_row_schema`].
+    pub fn remove_files(&mut self, remove_metadata: Box<dyn EngineData>) {
```

Comment:
I noticed that this is also addressed here: https://github.com/delta-io/delta-kernel-rs/pull/1020/files#diff-241320506790ddcf1809819f7905f4463afa829dc71b1c76710fb968c450623dR288

Maybe we can get that PR in first

Reply:
This is from the top of the stack which references that PR.  That PR never got merged so hopefully we can close it out on https://github.com/delta-io/delta-kernel-rs/pull/1242